### ISSUE TITLE
Remove classes.dot after running test suite has finished

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#4762]
### Bug description

The testing of `run_pyreverse` in `tests/test_pylint_runners.py` creates a `classes.dot` document during testing. Although it only takes a little effort to remove this document after the test suite is done, it would be good if it was automatically removed afterwards.

Something as simple as checking for `getattr(runner, "__name__") == "run_pyreverse")` on line 19 and then `os.remove(classes.dot)` would do the trick, although I am not sure if such a solution fits the code style/policy of the project.

### Command used

```shell
tox -e py39  -- -k pylint_runners
```

### Pylint output

```shell
Creates `classes.dot` in main directory.
```

### Expected behavior

Removing `classes.dot` after testing has been completed.

### Pylint version

```shell
pylint 2.9.5
astroid 2.6.5
Python 3.9.6 (default, Jun 29 2021, 05:25:02) 
[Clang 12.0.5 (clang-1205.0.22.9)]
```

### OS / Environment

macOS Big Sur

**Comment by Pierre-Sassoulas** [comment]
I agree, this has been slightly annoying me from some time.

**Comment by DanielNoord** [comment]
Should I create a PR as suggested in the OP? Or do you have another solution in mind?

**Comment by Pierre-Sassoulas** [comment]
I'd be glad if you open a PR for this :) I think we could use the pytest ``tmp_path`` fixture as removing it at the end of the test does not work if the tests fail or add a lot of boilerplate.
